crates/astria-sequencer/src/app.rs

```diff
@@ -237,7 +238,14 @@ impl App {
         self.is_proposer = true;
         self.update_state_for_new_round(&storage);
 
-        let (signed_txs, txs_to_include) = self.execute_block_data(prepare_proposal.txs).await;
+        let remaining_tx_bytes: usize = usize::try_from(prepare_proposal.max_tx_bytes)
+            .expect("should be able to convert max_tx_bytes to usize")
+            .checked_sub(std::mem::size_of::<GeneratedCommitments>())
+            .expect("commitment size should not be larger than the prepare proposal's max bytes");
+
+        let (signed_txs, txs_to_include) = self
+            .execute_and_filter_block_data(VecDeque::from(prepare_proposal.txs), remaining_tx_bytes)
+            .await;
```

Comment:
both this function and `execute_block_data` execute the txs, but we don't want to double-execute everything (this will also cause everything to fail on second execution) - tbh i preferred the previous solution as it also doesn't loop over all the txs twice. i don't think there's a way to split apart the length-check and execution unfortunately as we only include successfully-executing txs in a block. i'd suggest moving it back into one `filter_and_execute_block_data` function but breaking that function into multiple functions (ie inside the for loop, it calls a fn that handles just that tx).

Reply:
I think I'm missing something, I don't see where we're doing double execution